include/openPMD/Iteration.hpp

```diff
@@ -99,8 +136,33 @@ class Iteration : public Attributable
     void flush();
     void read();
 
+    enum class CloseStatus
+    {
+        Open,
+        ClosedInFrontend,
+        ClosedInBackend
+    };
+
+    /*
+     * An iteration may be logically closed in the frontend,
+     * but not necessarily yet in the backend.
+     * Will be propagated to the backend upon next flush.
+     * Store the current status.
+     */
+    std::shared_ptr< CloseStatus > m_closed =
+        std::make_shared< CloseStatus >( CloseStatus::Open );
+
+    /**
+     * @brief Verify that a closed iteration has not been wrongly accessed.
+     *
+     * @return true If closed iteration had no wrong accesses.
+     * @return false Otherwise.
+     */
+    bool
+    verifyClosed() const;
```

Comment:
I would rename this to `closeViolated()` or similar, `verifyClosed()` is not clear. Also depends on the return status if you call this on a still open iteration. What shall this return?